src/resolvers/git.cr

```diff
@@ -144,26 +144,30 @@ module Shards
       update_local_cache
       ref = git_ref(version)
 
-      if file_exists?(ref, SPEC_FILENAME)
-        capture("git show #{Process.quote("#{ref.to_git_ref}:#{SPEC_FILENAME}")}")
+      if file_exists?(ref, spec_filename)
+        capture("git show #{Process.quote("#{ref.to_git_ref}:#{spec_filename}")}")
       else
-        Log.debug { "Missing \"#{SPEC_FILENAME}\" for #{name.inspect} at #{ref}" }
+        Log.debug { "Missing \"#{spec_filename}\" for #{name.inspect} at #{ref}" }
         nil
       end
     end
 
+    def spec_filename
+      File.join(subdir, SPEC_FILENAME)
+    end
+
     private def spec_at_ref(ref : GitRef, commit) : Spec
       update_local_cache
 
-      unless file_exists?(ref, SPEC_FILENAME)
-        raise Error.new "No #{SPEC_FILENAME} was found for shard #{name.inspect} at commit #{commit}"
+      unless file_exists?(ref, spec_filename)
+        raise Error.new "No #{spec_filename} was found for shard #{name.inspect} at commit #{commit}"
       end
 
-      spec_yaml = capture("git show #{Process.quote("#{ref.to_git_ref}:#{SPEC_FILENAME}")}")
+      spec_yaml = capture("git show #{Process.quote("#{ref.to_git_ref}:#{spec_filename}")}")
       begin
         Spec.from_yaml(spec_yaml)
       rescue error : Error
-        raise Error.new "Invalid #{SPEC_FILENAME} for shard #{name.inspect} at commit #{commit}: #{error.message}"
+        raise Error.new "Invalid #{spec_filename} for shard #{name.inspect} at commit #{commit}: #{error.message}"
```

Comment:
I went ahead and changed all of these uses of the SPEC_FILENAME constant as I was working without considering the context but looking at it now, it seems like in these messages it would be better without the subdirectory name in it. 🤔